pkg/internal/pop/msal.go

```diff
@@ -59,6 +48,46 @@ func AcquirePoPTokenInteractive(
 	return result.AccessToken, result.ExpiresOn.Unix(), nil
 }
 
+// AcquirePoPTokenByUsernamePassword acquires a PoP token using MSAL's username/password login flow
+// This flow does not require user interaction as credentials have already been provided
+func AcquirePoPTokenByUsernamePassword(
```

Comment:
nit : Do you think we should split each flow into a separated file like it does inside the `token` folder ? instead of having all of them inside this `msal.go` file ?

Reply:
I separated by `confidential` and `public` - we can discuss further but I think this made the most sense since the `public` client methods share some functionality